### ISSUE TITLE
Chance level accuracy when training using gpu

### ISSUE THREAD

**Issue by tremblerz** [tensorflow/models#3806]
------------------------

### System information
- **What is the top-level directory of the model you are using**: `models/research/slim`
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: `No`
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: `Linux Ubuntu 16.04`
- **TensorFlow installed from (source or binary)**: `tried both`
- **TensorFlow version (use command below)**: `1.6`
- **Bazel version (if compiling from source)**: `0.11.1`
- **CUDA/cuDNN version**: `9.0/7.1`
- **GPU model and memory**: `Gtx 980`
- **Exact command to reproduce**: `python train_image_classifier.py --train_dir /tmp/train/ --dataset_name cifar10 --dataset_split train --dataset_dir /tmp/data/ --clone_on_cpu=True --model_name inception_v3`

### Describe the problem
When training models present in `research/slim/` with gpu, we face the problem of model not getting trained and giving chance level accuracy(0.1 on cifar10) after training for 2-3 epochs. We found this problem with multiple different cuda+cudnn combinations. This problem doesn't take place if training is being done on CPUs and gives the expected incremental accuracy with training time. Another related problem we have been experiencing since long time is related with parameter `--clone_on_cpu`, if it is kept False(default option) then even the heaviest models train at a very fast speed and give chance level accuracy when evaluated. We performed testing of same repository by training [CIFAR10 tutorial](https://github.com/tensorflow/models/tree/master/tutorials/image/cifar10) with all gpu configurations same as before and it worked well as expected.
